### PR TITLE
[QUIC] Connections

### PR DESCRIPTION
**Pull request by Mr3zee**
**Subsystem**
ktor-network

**Solution**
Foundation for usage of QUIC connections

### CODE REVIEW

File: ktor-network/ktor-network-quic/jvmAndNix/src/io/ktor/network/quic/connections/ConnectionID.kt [hunk 1]
```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright 2014-2023 JetBrains s.r.o and contributors. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package io.ktor.network.quic.connections
+
+internal typealias ConnectionID = ByteArray
```
Comment (e5l) [inline ktor-network/ktor-network-quic/jvmAndNix/src/io/ktor/network/quic/connections/ConnectionID.kt:7]: Maybe value class could be better

File: ktor-network/ktor-network-quic/jvmAndNix/src/io/ktor/network/quic/connections/ConnectionID.kt [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+internal infix fun ConnectionID?.neq(other: ConnectionID?): Boolean {
+    return !eq(other)
+}
+
+internal infix fun ConnectionID?.eq(other: ConnectionID?): Boolean {
+    if (this == null && other == null) return true
```
Comment (e5l) [inline ktor-network/ktor-network-quic/jvmAndNix/src/io/ktor/network/quic/connections/ConnectionID.kt:14]: ByteArray.contentEquals: https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/content-equals.html

**Review by e5l — APPROVED** [review]
lgtm in general. I'm not happy with many `TODO` stubs, but the PR is big, so it should be OK.

**Comment by Mr3zee** [comment]
> I'm not happy with many `TODO` stubs, but the PR is big, so it should be OK.

Yeah, most of them are stubs, some are just remainders for future me in the places, where I can overlook something important, I'll get rid of them in the upcoming reviews

File: ktor-network/ktor-network-quic/jvmAndNix/src/io/ktor/network/quic/connections/ConnectionID.kt (hunk 2 above)
Comment (Mr3zee) [reply ktor-network/ktor-network-quic/jvmAndNix/src/io/ktor/network/quic/connections/ConnectionID.kt:14]: oops

**Pull request merged**